### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Amazinggracee**
### Project requirements

- [ ] Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- Think about how you can use two decorators in order to capitalize and trim people's names.
- In the following steps you will get instructions for implementing the Decorator design pattern for this case.
- Interface
- Create a class Nameable.
- Implement a method called correct_name that will raise a NotImplementedError.

- [ ] Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- [ ] Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @nameable.

- [ ] Prepare CapitalizeDecorator and TrimmerDecorator

1. **For the CapitalizeDecorator:**
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.

2. **For the TrimmerDecorator:**
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: classes/decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (thecodechaser) [inline classes/decorator.rb]: - [ ] Please make sure the nameable variable is readable and writable to work the decorate pattern as expected.
  - You can consider using the code `attr_accessor :nameable`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Amazinggracee👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/decorator.rb (hunk 1 above)
Comment (Amazinggracee) [reply classes/decorator.rb]: Thank you for the suggestion.

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Amazinggracee, 





Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice fixing the suggested change 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**